rust-toolchain

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.65"
+channel = "nightly"
```

Comment:
We agreed to use the latest stable version. Using nightly will increase inconsistency in the project significantly, and I'm pretty sure it will slow down development.

Reply:
Ah this was a mistake, we need to build docs with nightly (many crates require nightly to build their docs, since this is what docs.rs does, so the build just fails on stable).

I changed this because I'm using the rustc provided by the devshell, which reads this file. I'll change it back, but if we want to run it in CI, we'll need nightly. However, if we pin to a specific nightly version, we'll avoid the issues with random breakage (since nix will pin it to a specific nightly version)

Reply:
I'd suggest fixing all test suits first, having stable builds, and then thinking about what to do with docs.
It's better to have consistency and production-ready product and docs with errors than to go to the path where we need to repeatedly re-do / re-fix some things.

Ps: tests are failing again now.

Reply:
We discussed this offline. We'll bring up this topic for the following tech discussion.